packages/frontend/src/components/ChartConfigPanel/FieldLayoutOptions.tsx

```diff
@@ -1,62 +1,143 @@
+
+    // X axis logic
+    const xAxisField = items.find(
+        (item) => getItemId(item) === dirtyLayout?.xField,
+    );
+
+    const onXClick = (itemId: any) => {
+        const isActive = xAxisField && getItemId(xAxisField) === itemId;
+        setXField(!isActive ? itemId : undefined);
+    };
+
+    // Y axis logic
+    const yActiveField = (field: string) => {
+        return items.find((item) => field.includes(item.name));
+    };
+
+    const availableYFields = () => {
+        return items.filter(
+            (item) => !dirtyLayout?.yField?.includes(getItemId(item)),
+        );
+    };
+
+    const onAddField = () => {
+        addSingleSeries(getItemId(availableYFields()[0]));
+    };
+
+    // Group series logic
+    const groupSelectedField = items.find(
+        (item) => getItemId(item) === pivotDimension,
+    );
+
+    const onGroupClick = (itemId: any) => {
+        const isGroupActive = !!pivotDimension && pivotDimension === itemId;
+
+        return !isGroupActive
+            ? setPivotDimensions([itemId])
+            : setPivotDimensions(undefined);
+    };
+
     return (
         <>
-            <GridFieldLabel>
-                <Icon icon={getItemIcon(item)} color={getItemColor(item)} />
-                <span>
-                    {isField(item) ? `${item.tableLabel} ` : ''}
-                    <b>{isField(item) ? item.label : item.displayName}</b>
-                </span>
-            </GridFieldLabel>
-            <ButtonGroup>
-                <Button
-                    intent={isXActive ? 'primary' : 'none'}
-                    onClick={() => onXClick(!isXActive)}
-                >
-                    x
-                </Button>
-                <Button
-                    intent={isYActive ? 'primary' : 'none'}
-                    onClick={() => onYClick(!isYActive)}
-                >
-                    y
-                </Button>
-                <Button
-                    intent={isGroupActive ? 'primary' : 'none'}
-                    onClick={() => onGroupClick(!isGroupActive)}
-                >
-                    group
-                </Button>
-            </ButtonGroup>
+            <AxisGroup>
+                <AxisTitle>X axis field</AxisTitle>
+                <AxisFieldDropdown>
+                    <FieldAutoComplete
+                        fields={items}
+                        activeField={xAxisField}
+                        onChange={(item) => {
+                            if (isField(item)) {
+                                onXClick(getItemId(item));
+                            }
+                        }}
+                    />
+                </AxisFieldDropdown>
+            </AxisGroup>
+            <AxisGroup>
+                <AxisTitle>Y axis field</AxisTitle>
+
+                {dirtyLayout?.yField
+                    ? dirtyLayout?.yField.map((field, index) => (
+                          <AxisFieldDropdown key={`${index}-y-axis`}>
```

Comment:
Using the index as a key is a bad practice, we can use instead the actual field var since it should be unique in that array.